Version 1 release

NeHo4y opened the pull request.

dxahtepb committed:
Introduce User model and service (#1)

* Introduce User domain entity and User CRUD repository

* Introduce basic UserService

For now we can only create users and log them in (more like just find them with username-password pair). This functionality are likely to be reworked in later commits, tho.

* Add function to find user by its username to UserService

* Add function to delete user to UserService

* Add phone number to the user domain model

* Save creation date for user entity

* Add service to update user info

For now there is only one editable column - user's phone number.

* Filter deleted users out from repository's find methods

* Return optional from find method in repository

Do it this way to be more consistent with findById and other default methods

NeHo4y committed:
Implement File Service (#5)

Implement File Service

maxxkut committed:
Implement Category, Topic, Subtopic, Feedback (#4)

Introducing Category, Topic, Subtopic, Feedback

NeHo4y committed:
Implement Comment Service (#3)

Implement Comment

NeHo4y committed:
Add exceptions to all current services (#8)

* Add custom exceptions to file, feedback, category
* Change tests, add exception to comment, remove unused imports

Co-authored-by: Andrey Chausov <dxahtepb@gmail.com>

dxahtepb committed:
Implement Follower model and service (#2)

Implemented service for managing which users are associated with a feedback as a followers. For now there is two types of followage: as an assignee or as a watcher. This service may be used to send a notifiaction to watchers on new comments, for instance.

dxahtepb committed:
Add Ktlint to Gradle jobs and to Github Actions CI (#9)

Also apply ktlint auto-formatting, although some issues required manual fixes (i.e. bad file names). argument-list-wrapping rule have a bug which is not fixed in the current version, so it is disabled. Tbh, it doesn't comply with Kotlin official code-style anyway.

dxahtepb commented:
Still, I want the Notification service (#6) in this release. What do u guys think about it?

NeHo4y commented:
I think this may work

dxahtepb committed:
Notification service (#6)

Introduce a basic Notification service which is capable of sending different types of messages
